### ISSUE TITLE
Introduce a Ballerina connector for SendGrid REST API

### ISSUE THREAD

**Issue by NipunaRanasinghe** [ballerina-platform/ballerina-library#7041]
We are planning to introduce a new Ballerina connector for the latest SendGrid REST API, generated using it's OpenAPI specification.

Related links:
- https://docs.sendgrid.com/api-reference
- https://github.com/twilio/sendgrid-oai

This includes the following tasks:
1. Create a new repository under 'ballerina-platform' organization.
2. Review and sanitize the OpenAPI specification.
3. Generate the connector client.
4. Implement a test suite covering the core functionalities.
5. Write documentation with examples.

**Comment by shatanikmahanty** [comment]
Hi @NipunaRanasinghe I would like to work on this. I can start in 4 days from now!

**Comment by MohamedSabthar** [comment]
👋 Welcome, @shatanikmahanty! 🚀

We're thrilled to have you join the [Ballerina Hacktoberfest](https://ballerina.io/hacktoberfest/) community! The issue has been assigned to you, and we’re excited to see your contributions.

To help you get started, here are a few essential resources:

1. **Connector Contributor Guide**: Make sure to go through our [Ballerina Hacktoberfest connector contributor guidelines](https://ballerina.io/hacktoberfest/connector-contributor-guide/) and follow the exact steps, to contribute effectively.
2. **Learn Ballerina**: Whether you're new to Ballerina or looking to enhance your skills, check out our [official learning resources](https://ballerina.io/learn/).
3. **Community & Support**: If you have technical questions, feel free to ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/ballerina) with the Ballerina tag, or join us on [Discord](https://discord.com/invite/ballerinalang) to connect with other community members.

No contribution is too small, and your feedback is always welcome! Don’t hesitate to ask questions, propose new ideas, or report issues.

We are currently in the process of creating a GitHub repository for this connector module and will update you once it’s available. In the meantime, please go through the relevant resources and documentation related to the connector.

**Happy coding** and welcome aboard! 🎉

**Comment by tharindu432** [comment]
hi, if the assigned person will not be able to start on this issue, can you assign this to me @MohamedSabthar ?

**Comment by shatanikmahanty** [comment]
HI @tharindu432, I was waiting for the team to create a GitHub repo as they mentioned. But if you want to contribute, if they assign please go ahead as I am a bit busy these days with work. Thanks!

**Comment by ayeshLK** [comment]
Thanks for notifying @shatanikmahanty 

Hi @tharindu432, it looks like you’re currently working on the Ballerina LinkedIn connector [1]. I just wanted to check if you've had time to complete it; if not, it might be a good idea to let someone else take on this task. Let me know what you think.

[1] - https://github.com/ballerina-platform/ballerina-library/issues/7056

**Comment by tharindu432** [comment]
Yes, i am currently doing the Linkedin connector implementation. I put this comment before it . You can assign this to someone else.😊

**Comment by ayeshLK** [comment]
👋 Welcome, @ravinduheshanweerasinghe! 🚀

We're thrilled to have you join the [Ballerina Hacktoberfest](https://ballerina.io/hacktoberfest/) community! The issue has been assigned to you, and we’re excited to see your contributions.

To help you get started, here are a few essential resources:

1. **Connector Contributor Guide**: Make sure to go through our [Ballerina Hacktoberfest connector contributor guidelines](https://ballerina.io/hacktoberfest/connector-contributor-guide/) and follow the exact steps, to contribute effectively.
2. **Learn Ballerina**: Whether you're new to Ballerina or looking to enhance your skills, check out our [official learning resources](https://ballerina.io/learn/).
3. **Community & Support**: If you have technical questions, feel free to ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/ballerina) with the Ballerina tag, or join us on [Discord](https://discord.com/invite/ballerinalang) to connect with other community members.

No contribution is too small, and your feedback is always welcome! Don’t hesitate to ask questions, propose new ideas, or report issues.

We are currently in the process of creating a GitHub repository for this connector module and will update you once it’s available. In the meantime, please go through the relevant resources and documentation related to the connector.

**Happy coding** and welcome aboard! 🎉

**Comment by ayeshLK** [comment]
The repository for the **Ballerina SendGrid Connector** is now live and ready for development! 🎉 You can find the project structure initialized [here](https://github.com/ballerina-platform/module-ballerinax-sendgrid).

Feel free to dive in and start building the connector! If you need any assistance or have any questions, just drop a comment here — we’re happy to help! 😊

Happy coding and enjoy Hacktoberfest! 🚀👨‍💻👩‍💻

**Comment by ayeshLK** [comment]
Hey @ravinduheshanweerasinghe!
Hope everything’s going well so far! Just a quick update on the project deadlines: it's set for one month from the day you were officially assigned. But if you encounter any challenges or unforeseen issues, please feel free to reach out, and we can consider adjusting the deadline accordingly.

If you have any questions or need help with anything, please feel free to reach out. Keep up the awesome work! 🚀

**Comment by ayeshLK** [comment]
Hey @ravinduheshanweerasinghe!

Just checking the progress of the connector development, were you able to start the development of the connector ? If not please take your time and go through the SendGrid API documentation before starting the development. The first step is to identify the OpenAPI contract and generate the connector. 

Please feel free to reach out if you have any concerns or questions.

Happy Coding 🚀

**Comment by ayeshLK** [comment]
Hey @ravinduheshanweerasinghe greetings 👋 

Just checking in on the progress of the SendGrid connector development.

Any update from your side ? Were you able to generate the connector using the provided resources [1][2] ?

[1] - https://docs.sendgrid.com/api-reference
[2] - https://github.com/twilio/sendgrid-oai

**Comment by ayeshLK** [comment]
Hey @ravinduheshanweerasinghe 👋 

Just checking on the progress of the connector project. 

Are you actively working on this project ? If not we will have to reassign this to someone else.

**Comment by ayeshLK** [comment]
Since there is no active contribution we will unassign the issue.
